src/Microsoft.DotNet.ServiceFabric.ServiceHost/TemporaryFiles.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Fabric;
+using System.IO;
+
+namespace Microsoft.DotNet.ServiceFabric.ServiceHost
+{
+    public class TemporaryFiles : IDisposable
+    {
+        private string _isolatedTempPath;
+
+        public TemporaryFiles(ServiceContext context)
+        {
+            ServiceContext context1 = context;
+            _isolatedTempPath = Environment.ExpandEnvironmentVariables($"%TEMP%\\{context1.ServiceTypeName}\\{context1.ReplicaOrInstanceId}");
+        }
+
+        public void Initialize()
+        {
+            // Do cleanup in the case of unhealthy exit last time.
+            Cleanup();
+            Directory.CreateDirectory(_isolatedTempPath);
+        }
+
+        public string GetFilePath(params string[] parts)
+        {
+            return Path.Combine(_isolatedTempPath, Path.Combine(parts));
+        }
+
+        public void Dispose()
+        {
+            Cleanup();
+        }
+
+        private void Cleanup()
+        {
+            try
+            {
+                // Try and delete/clean it
+                if (Directory.Exists(_isolatedTempPath))
+                    Directory.Delete(_isolatedTempPath, true);
+            }
+            catch (IOException)
+            {
+                // It might not be here, it might be locked... there isn't really anything interesting to do, move on
```

Comment:
Or are files then in a different directory for each deployment? And if, how will this erase the old deployment if deployment ID (replica/instance whatever it is called) is part of the path?

Reply:
The old deployment stops running before the new deployment starts, so there is no possibility of messing up a running instance.

Reply:
Oh good. I was going to ask alex this very question, and he preemptivly answered. :-D

Reply:
Though I added a logger anyway. I didn't have one originally because it was a static method, but now that it's injectable, I can easily inject and use an ILogger to log the exception if there is one.